compiler/rustc_infer/src/infer/error_reporting/mod.rs

```diff
@@ -1484,13 +1484,14 @@ impl<'a, 'tcx> InferCtxt<'a, 'tcx> {
                 for (key, values) in types.iter() {
                     let count = values.len();
                     let kind = key.descr();
+                    let mut returned_async_output_error = false;
                     for sp in values {
                         err.span_label(
                             *sp,
                             format!(
                                 "{}{}{} {}{}",
-                                if sp.is_desugaring(DesugaringKind::Async) {
-                                    "the `Output` of this `async fn`'s "
+                                if sp.is_desugaring(DesugaringKind::Async) && returned_async_output_error == false {
+                                    "checked the `Output` of this `async fn`, "
```

Comment:
I was never happy with the wording of this sentence when I wrote it originally 😅

What this was meant to say succinctly was "we found an opaque type, that opaque type we found is the return type of this async function, this async function return type desugars to `impl Future<Output = u32>`"

I think the change now changes the meaning slightly, but I don't know what a better wording might be :(

Reply:
I agree the wording could be better...but am also not sure how I would form it. I do feel it is a little easier for the user to understand what the compiler did with this wording, and therefore it's a little easier for the user to understand what action they need to take.

Reply:
Yeah, looking at it with fresh eyes, the new wording isn't _great_ but it is very reasonable.